### PR TITLE
Issue #462: `checkout_github_prs()` returns empty list when no PRs for the given criteria are found

### LINKED ISSUE #462: `checkout_github_prs()` should NOT fail if no PRs for the given criteria are found

### PR DESCRIPTION
**Pull request by amalota**
closes #462

### CODE REVIEW

File: git_machete/client.py [hunk 1]
```diff
@@ -1950,25 +1950,28 @@ def __get_valid_pull_requests(prs_list: Optional[List[int]],
...
+                            warn(f"PR #{pr_no} is not found in repository `{org}/{repo}`")
+                            return []
             if not result:
                 raise MacheteException(
                     f"Given PRs: {', '.join(map(str, prs_list))} are not found in repository `{org}/{repo}`")
             return result
         if all:
             if not all_opened_prs_from_github:
-                raise MacheteException(f"Currently there is not any pull request opened in repository `{org}/{repo}`")
+                warn(f"Currently there is not any pull request opened in repository `{org}/{repo}`")
```
Comment (PawelLipski) [inline git_machete/client.py]:
```suggestion
                warn(f"Currently there are no pull requests opened in repository `{org}/{repo}`")
```

File: git_machete/client.py [hunk 2]
```diff
@@ -1950,25 +1950,28 @@ def __get_valid_pull_requests(prs_list: Optional[List[int]],
                         if len(prs_list) > 1:
                             warn(f"PR #{pr_no} is not found in repository `{org}/{repo}`, skipping.")
                         else:
-                            raise MacheteException(f"PR #{pr_no} is not found in repository `{org}/{repo}`")
+                            warn(f"PR #{pr_no} is not found in repository `{org}/{repo}`")
```
Comment (PawelLipski) [inline git_machete/client.py]: For this case, I'd leave an error... a user expected a specific PR number to be present, and this particular PR does not exist

Comment (PawelLipski) [reply git_machete/client.py]: As opposed to the other cases, where zero open PRs are a kinda expected/possible case

File: git_machete/client.py [hunk 3]
```diff
@@ -1950,25 +1950,28 @@ def __get_valid_pull_requests(prs_list: Optional[List[int]],
                         if len(prs_list) > 1:
                             warn(f"PR #{pr_no} is not found in repository `{org}/{repo}`, skipping.")
```
Comment (PawelLipski) [inline git_machete/client.py]: I wouldn't make a difference in handling of the `len(prs_list) > 1` case, let's fail this one as well

File: git_machete/client.py [hunk 4]
```diff
@@ -1950,25 +1950,28 @@ def __get_valid_pull_requests(prs_list: Optional[List[int]],
...
-                raise MacheteException(
-                    f"Current user {user} has no open pull request in repository `{org}/{repo}`")
+                warn(f"Current user {user} has no open pull request in repository `{org}/{repo}`")
+                return []
             return result
         elif by:
             result = [pr for pr in all_opened_prs_from_github if pr.user == by]
             if not result:
-                raise MacheteException(f"User {by} has no open pull request in repository `{org}/{repo}`")
+                warn(f"User {by} has no open pull request in repository `{org}/{repo}`")
```
Comment (PawelLipski) [inline git_machete/client.py]:
```suggestion
                warn(f"User `{by}` has no open pull request in repository `{org}/{repo}`")
```

Comment (PawelLipski) [reply git_machete/client.py]: Underline for the sake of avoiding some tricky usernames which might look like a regular part of sentence and not a username

Comment (PawelLipski) [reply git_machete/client.py]: `User that has no pull request...` etc. :trollface:

File: git_machete/client.py [hunk 5]
```diff
@@ -1950,25 +1950,28 @@ def __get_valid_pull_requests(prs_list: Optional[List[int]],
...
-                raise MacheteException(f"Currently there is not any pull request opened in repository `{org}/{repo}`")
+                warn(f"Currently there is not any pull request opened in repository `{org}/{repo}`")
+                return []
             return all_opened_prs_from_github
         elif my and user:
             result = [pr for pr in all_opened_prs_from_github if pr.user == user]
             if not result:
-                raise MacheteException(
-                    f"Current user {user} has no open pull request in repository `{org}/{repo}`")
+                warn(f"Current user {user} has no open pull request in repository `{org}/{repo}`")
```
Comment (PawelLipski) [inline git_machete/client.py]:
```suggestion
                warn(f"Current user `{user}` has no open pull request in repository `{org}/{repo}`")
```

File: git_machete/tests/functional/test_machete.py [hunk 6]
```diff
@@ -2409,12 +2409,8 @@ def test_github_checkout_prs(self) -> None:
         repo: str
         org: str
         (org, repo) = get_parsed_github_remote_url(self.repo_sandbox.remote_path)
-        expected_error_message = f"PR #100 is not found in repository `{org}/{repo}`"
-        with self.assertRaises(MacheteException) as e:
-            self.launch_command('github', 'checkout-prs', '100')
-        if e:
-            self.assertEqual(e.exception.parameter, expected_error_message,
-                             'Verify that expected error message has appeared when given pull request to checkout does not exists.')
+        expected_msg = f"Warn: PR #100 is not found in repository `{org}/{repo}`\n"
```
Comment (PawelLipski) [inline git_machete/tests/functional/test_machete.py]: Pls add a test case for where `--mine`/`--by` would behave differently before & after this change in implementation

File: git_machete/client.py [hunk 7]
```diff
@@ -1948,7 +1948,7 @@ def __get_valid_pull_requests(prs_list: Optional[List[int]],
                         result.append(pr_from_github)
                     else:
                         if len(prs_list) > 1:
-                            warn(f"PR #{pr_no} is not found in repository `{org}/{repo}`, skipping.")
+                            raise MacheteException(f"PR #{pr_no} is not found in repository `{org}/{repo}`, skipping.")
```
Comment (amalota) [inline git_machete/client.py]: @PawelLipski Looking at this now i wouldn't raise and error here:
1. There is no difference between `if` and `else` cases, both raise exception, just different msg
2. `if len(prs_list) > 1:` case doesn't make sense anymore with, `skipping'` word in the msg (it doesn't get skipped anymore, the function fails).

To sum up i would leave it as it was before:
```
if len(prs_list) > 1:
    raise MacheteException(f"PR #{pr_no} is not found in repository `{org}/{repo}`, skipping.")
```
or remove `if` and `else` and just do `raise the raise MacheteException(f"PR #{pr_no} is not found in repository `{org}/{repo}`")`

Comment (PawelLipski) [reply git_machete/client.py]: Okay! let's go for the latter (withOUT `if`/`else`) and see what's the resulting UX

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**